src/toolkit/position.cycpp.h

```diff
@@ -28,5 +25,14 @@ std::vector<int> cycpp_shape_latitude = 0;
        "be expressed in degrees as a double." \
 }
 double longitude;
+
+protected:
+
+  void InitializePosition(cyclus::Agent* agent) {
```

Comment:
This injected function becomes a member function of the agent, so we probably don't need to pass in an agent pointer an can just assume `this` and pass that to `RecordPosition`